src/lib.rs

```diff
@@ -360,75 +377,101 @@ pub use crate::item::{
 };
 
 #[cfg(feature = "full")]
+#[cfg_attr(syn_enable_doc_cfg, doc(cfg(feature = "full")))]
 mod file;
 #[cfg(feature = "full")]
+#[cfg_attr(syn_enable_doc_cfg, doc(cfg(feature = "full")))]
 pub use crate::file::File;
 
+#[cfg(any(feature = "full", feature = "derive"))]
+#[cfg_attr(syn_enable_doc_cfg, doc(cfg(any(feature = "full", feature = "derive"))))]
 mod lifetime;
+#[cfg(any(feature = "full", feature = "derive"))]
+#[cfg_attr(syn_enable_doc_cfg, doc(cfg(any(feature = "full", feature = "derive"))))]
 pub use crate::lifetime::Lifetime;
```

Comment:
Note, that I added the feature-cfgs for the lifetime module. The doc-comments of `Lifetime` suggested, that this was intended.

Reply:
There is another CI failure when running `cargo build --no-default-features --features 'proc-macro parsing printing'` due to the `Lifetime` struct being used inside the `buffer` module, which is enabled on the `parsing` feature.

I'd suggest to add `, feature = "parsing"` to the `any()`-expression and update the doc comment of `Lifetime`.

Whats your thoughts on this?